src/Evolution/Systems/GeneralizedHarmonic/Characteristics.hpp

```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace Tags {
+template <typename Tag>
+struct Normalized;
+}  // namespace Tags
+/// \endcond
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace GeneralizedHarmonic {
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Compute the characteristic speeds for the generalized harmonic system.
+ *
+ * Computes the speeds as described in "A New Generalized Harmonic
+ * Evolution System" by Lindblom et. al, https://arxiv.org/abs/gr-qc/0512093.
+ * The characteristic fields' names used here differ from this paper:
+ *
+ * \f{align*}
+ * \mathrm{SpECTRE} && \mathrm{Lindblom} \\
+ * u^{\psi}_{ab} && u^0_{ab} \\
+ * u^{\pm}_{ab} && u^{1\pm}_{ab} \\
+ * u^0_{iab} && u^2_{iab}
+ * \f}
+ *
+ * The speeds \f$v\f$, are given by:
+ * \f{align*}
+ * v_{\psi} =& -(1 + \gamma_1) n_k N^k \\
+ * v_{0} =& -n_k N^k \\
+ * v_{\pm} =& -n_k N^k \pm N
+ * \f}
+ *
+ * where \f$N, N^k\f$ are the lapse and shift respectively, \f$\gamma_1\f$ is a
+ * constraint damping parameter, and \f$n_k\f$ is the unit normal to the
```

Comment:
We still seem to be using `n` as the normal for at least 2 different things, maybe 3. In this context we can use `t_k`

Reply:
I had thought that `n` was pretty consistently used to denote a spatial normal one-form. I also interpret `t` as being a spacetime normal to a hypersurface.

Reply:
ah yep, I was mistaken, sorry